common/content-api.js

```diff
@@ -47,15 +48,14 @@ function fetchAllLocales(toUrlPathFn, options = {}) {
  * Adds the preview parameters to the request
  * (if accessed via the preview domain)
  */
-function addPreviewParams(previewMode, params) {
-    if (!previewMode) {
-        return params;
-    }
-
-    let previewParams = {};
-    previewParams[previewMode.mode] = previewMode.id; // eg. ?draft=123
-
-    return Object.assign({}, previewParams, params);
+function addPreviewParams(requestParams = {}, params = {}) {
+    const globalParams = pick(requestParams, [
+        'social',
+        'x-craft-live-preview',
+        'x-craft-preview',
+        'token'
```

Comment:
I couldn't put the locale in here as it's not a query param, but if we passed around the whole `req` object I guess we could? But this seems complex as we do some unusual things with locales for some endpoints (eg. fetching both or forcing one language etc).